src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java

```diff
@@ -377,6 +383,49 @@ protected FinishTestItemRQ buildFinishStepRq(FrameworkMethod method, String stat
 		return parameters.isEmpty() ? null : parameters;
 	}
 
+	protected Optional<Integer> getTestCaseId(FrameworkMethod method, Object runner, String codeRef) {
+		if (!(method.isStatic() || isIgnored(method))) {
+			Object target = LifecycleHooks.getTargetForRunner(runner);
+			return ofNullable(method.getMethod().getDeclaredAnnotation(TestCaseId.class)).flatMap(annotation -> {
+				if (annotation.isParameterized() && target instanceof ArtifactParams) {
+					return getParameterizedTestCaseId(target, codeRef);
+				} else {
+					return Optional.of(annotation.value());
+				}
+			});
+
+		}
+		return Optional.of(Objects.hashCode(codeRef));
+	}
+
+	protected Optional<Integer> getParameterizedTestCaseId(Object target, String codeRef) {
+
+		com.google.common.base.Optional<Map<String, Object>> params = ((ArtifactParams) target).getParameters();
```

Comment:
maybe java.util.Optional?

Reply:
JUnit ArtifactParams returns guava's Optional, so it's not possible